### ISSUE TITLE
Crash / Imcompatibility with Immersive Portals

### ISSUE THREAD

**Issue by MightyKnight** [akoimeexx/mimics-mod#10]
## What is happening?
My world crashes on load / after getting killed by a mimic. I have been able to track this down to only happen if Immersive Portals is installed alongside.

### Steps to reproduce:
0. Make sure to have Mimics and Immersive Portals installed
1. Create a world
2. Use the command `/summon mimics:mimic`. You will receive an error message _An error occured during the execution of the command_ of type NullPointerException. (see picture below)
![grafik](https://user-images.githubusercontent.com/38622942/119053645-a4886f00-b9c6-11eb-9f55-ce7d7056ce12.png)
3. Kill yourself, for example with `/kill`.
4. Click the respawn button.

:x: Upon respawn your game will crash.

## Further info
Using only these 3 mods with Fabric Loader 0.11.3:
- Fabric API 0.34.2
- Mimics 0.1.3
- Immersive Portals 0.90

## Crash Report

```
---- Minecraft Crash Report ----
// My bad.

Time: 20.05.21 23:48
Description: Ticking memory connection

java.lang.NullPointerException: Ticking memory connection
	at net.minecraft.class_2540.method_10788(class_2540.java:421)
	at net.minecraft.class_2540.method_10814(class_2540.java:417)
	at net.minecraft.class_2943$14.method_12723(class_2943.java:81)
	at net.minecraft.class_2943$14.method_12715(class_2943.java:78)
	at net.minecraft.class_2945.method_12782(class_2945.java:194)
	at net.minecraft.class_2945.method_12787(class_2945.java:139)
	at net.minecraft.class_2739.method_11052(class_2739.java:37)
	at com.qouteall.hiding_in_the_bushes.MyNetwork.createRedirectedMessage(MyNetwork.java:109)
	at net.minecraft.class_3244.localvar$zil000$modifyPacket(class_3244.java:3628)
	at net.minecraft.class_3244.method_14369(class_3244.java)
	at net.minecraft.class_3244.method_14364(class_3244.java:1130)
	at com.qouteall.immersive_portals.network.CommonNetwork.sendRedirectedPacket(CommonNetwork.java:51)
	at net.minecraft.class_3231.md71800c$lambda$injectSendpacketsOnStartTracking$0$0(class_3231.java:1055)
	at net.minecraft.class_3231$$Lambda$5491/2039909129.accept(Unknown Source)
	at net.minecraft.class_3231.method_18757(class_3231.java:229)
	at net.minecraft.class_3231.handler$zij000$injectSendpacketsOnStartTracking(class_3231.java:1055)
	at net.minecraft.class_3231.method_18760(class_3231.java:214)
	at net.minecraft.class_3898$class_3208.md71800c$lambda$updateEntityTrackingStatus$0$2(class_3898.java:1960)
	at net.minecraft.class_3898$class_3208$$Lambda$5485/1187675449.run(Unknown Source)
	at com.qouteall.immersive_portals.network.CommonNetwork.withForceRedirect(CommonNetwork.java:28)
	at net.minecraft.class_3898$class_3208.updateEntityTrackingStatus(class_3898.java:1957)
	at net.minecraft.class_3898.updateEntityTrackersAfterSendingChunkPacket(class_3898.java:2273)
	at com.qouteall.immersive_portals.chunk_loading.ChunkDataSyncManager.sendChunkDataPacketNow(ChunkDataSyncManager.java:88)
	at com.qouteall.immersive_portals.chunk_loading.ChunkDataSyncManager.onBeginWatch(ChunkDataSyncManager.java:45)
	at com.qouteall.immersive_portals.chunk_loading.ChunkDataSyncManager$$Lambda$2417/371346392.accept(Unknown Source)
	at com.qouteall.immersive_portals.my_util.SignalBiArged.lambda$connectWithWeakRef$1(SignalBiArged.java:35)
	at com.qouteall.immersive_portals.my_util.SignalBiArged$$Lambda$2418/720164260.accept(Unknown Source)
	at com.qouteall.immersive_portals.my_util.SignalBiArged.lambda$emit$0(SignalBiArged.java:18)
	at com.qouteall.immersive_portals.my_util.SignalBiArged$$Lambda$4840/1512685628.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.qouteall.immersive_portals.my_util.SignalBiArged.emit(SignalBiArged.java:18)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph.lambda$null$8(NewChunkTrackingGraph.java:180)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph$$Lambda$5472/1945825872.run(Unknown Source)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph.updateWatchingStatus(NewChunkTrackingGraph.java:67)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph.lambda$updatePlayerForChunkLoader$9(NewChunkTrackingGraph.java:174)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph$$Lambda$5470/1669166188.consume(Unknown Source)
	at com.qouteall.immersive_portals.chunk_loading.ChunkLoader.foreachChunkPos(ChunkLoader.java:45)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph.updatePlayerForChunkLoader(NewChunkTrackingGraph.java:168)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph.lambda$updateForPlayer$5(NewChunkTrackingGraph.java:154)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph$$Lambda$5468/852627755.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:175)
	at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:419)
	at com.google.common.collect.CollectSpliterators$1FlatMapSpliterator.lambda$forEachRemaining$1(CollectSpliterators.java:246)
	at com.google.common.collect.CollectSpliterators$1FlatMapSpliterator$$Lambda$5469/438750097.accept(Unknown Source)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at com.google.common.collect.CollectSpliterators$1FlatMapSpliterator.forEachRemaining(CollectSpliterators.java:246)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.qouteall.immersive_portals.chunk_loading.NewChunkTrackingGraph.updateForPlayer(NewChunkTrackingGraph.java:154)
	at net.minecraft.class_3324.handler$zhi000$onOnPlayerConnect(class_3324.java:2503)
	at net.minecraft.class_3324.method_14570(class_3324.java:273)
	at net.minecraft.class_3248.method_14384(class_3248.java:113)
	at net.minecraft.class_3248.redirect$zfo000$handlePlayerJoin(class_3248.java:562)
	at net.minecraft.class_3248.method_18785(class_3248.java:65)
	at net.minecraft.class_2535.method_10754(class_2535.java:228)
	at net.minecraft.class_3242.method_14357(class_3242.java:173)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:882)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:811)
	at net.minecraft.class_1132.method_3748(class_1132.java:91)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:670)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257)
	at net.minecraft.server.MinecraftServer$$Lambda$4832/1794418722.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 933813512 bytes (890 MB) / 2046820352 bytes (1952 MB) up to 2147483648 bytes (2048 MB)
	CPUs: 12
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		cloth-basic-math: Cloth Basic Math 0.5.1
		cloth-config2: Cloth Config v4 4.11.15
		fabric: Fabric API 0.34.1+1.16
		fabric-api-base: Fabric API Base 0.3.0+c88702897d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.0.0+dc716ea17d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.1+ca58154a7d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+ca58154a7d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.1+351679a77d
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+ca58154a7d
		fabric-containers-v0: Fabric Containers (v0) 0.1.11+9354966b7d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.1+ca58154a7d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.3+ca58154a7d
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.6+61336ba97d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.1.0+c88702897d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.2+ca58154a7d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+ca58154a7d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.6+ca58154a7d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.1+ca58154a7d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.7+2868a2287d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+9354966b7d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.1+ca58154a7d
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.1+ca58154a7d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.2+ca58154a7d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+ca58154a7d
		fabric-models-v0: Fabric Models (v0) 0.3.0+bc7a746f7d
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.2+d6244c6d7d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.8+ca58154a7d
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+ca58154a7d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.9.5+c88702897d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.2+ca58154a7d
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+ca58154a7d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.4+ca58154a7d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.2+2868a2287d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.4+ca58154a7d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.3.0+2868a2287d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+ca58154a7d
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+ca58154a7d
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+ca58154a7d
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.0+2868a2287d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.6+2868a2287d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.0+c045166c7d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.5+9354966b7d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.4+ca58154a7d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.1+ca58154a7d
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+ca58154a7d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.7+8183d3957d
		fabricloader: Fabric Loader 0.11.3
		imm_ptl_core: Immersive Portals Core 0.90
		immersive_portals: Immersive Portals 0.90
		java: Java HotSpot(TM) 64-Bit Server VM 8
		mimics: Mimics 0.1.3
		minecraft: Minecraft 1.16.5
		modmenu: Mod Menu v1.16.8+unknown.bd9230a
	Player Count: 1 / 8; [class_3222['PLAYERNAME'/121, l='ServerWorld minecraft:overworld Immersive', x=219.50, y=64.00, z=-247.50]]
	Data Packs: vanilla, Fabric Mods
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fabric'
```

**Comment by akoimeexx** [comment]
Ticking memory generally indicates you need more RAM allocated. Please increase the amount of memory allocated to your server and try again and report results here.

**Edit:** I should note, I am setting up a test env. for this to confirm the NPE portion.

**Comment by akoimeexx** [comment]
It's definitely something IP is doing, I get this from the crash instead of the ticking memory error:

```
[19:13:49] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.NullPointerException: Cannot invoke "java.lang.String.getBytes(java.nio.charset.Charset)" because "☃" is null
	at net.minecraft.class_2540.method_10788(class_2540.java:421) ~[intermediary-minecraft-1.16.5-client.jar:?]
	at net.minecraft.class_2540.method_10814(class_2540.java:417) ~[intermediary-minecraft-1.16.5-client.jar:?]
	at net.minecraft.class_2943$14.method_12723(class_2943.java:81) ~[intermediary-minecraft-1.16.5-client.jar:?]
	at net.minecraft.class_2943$14.method_12715(class_2943.java:78) ~[intermediary-minecraft-1.16.5-client.jar:?]
	at net.minecraft.class_2945.method_12782(class_2945.java:194) ~[intermediary-minecraft-1.16.5-client.jar:?]
	at net.minecraft.class_2945.method_12787(class_2945.java:139) ~[intermediary-minecraft-1.16.5-client.jar:?]
	at net.minecraft.class_2739.method_11052(class_2739.java:37) ~[intermediary-minecraft-1.16.5-client.jar:?]
	at com.qouteall.hiding_in_the_bushes.MyNetwork.createRedirectedMessage(MyNetwork.java:109) ~[8faa71d7-8707-457e-a384-a48e7b6d1c04.jar:?]
```

**Comment by akoimeexx** [comment]
Cursory look into @quoteall 's repos, I do not see any project where they have MyNetwork defined. Please report this issue on their tracker for IP.

**Comment by akoimeexx** [comment]
As an aside, I just fixed the bug from #9 and re-tested IP with that build on a whim which worked, so it looks like an issue when a tracked data point is nullable.

**Comment by MightyKnight** [comment]
Hi Akoi,
works like a charm with the new version. Thank you for fixing this, and that even so fast.
